core/chains/evm/logpoller/log_poller.go

```diff
@@ -881,22 +887,53 @@ func (lp *logPoller) PollAndSaveLogs(ctx context.Context, currentBlockNumber int
 	}
 }
 
+// Returns information about latestBlock, latestFinalizedBlockNumber
+// If finality tag is not enabled, latestFinalizedBlockNumber is calculated as latestBlockNumber - lp.finalityDepth (configured param)
+// Otherwise, we return last finalized block number returned from chain
+func (lp *logPoller) latestBlocks(ctx context.Context) (*evmtypes.Head, int64, error) {
+	// If finality is not enabled, we can only fetch the latest block
+	if !lp.useFinalityTag {
+		// Example:
+		// finalityDepth = 2
+		// Blocks: 1->2->3->4->5(latestBlock)
+		// lastFinalizedBlockNumber would be 3
+		latestBlock, err := lp.ec.HeadByNumber(ctx, nil)
+		if err != nil {
+			return nil, 0, err
+		}
+		// If chain has fewer blocks than finalityDepth, return 0
+		return latestBlock, mathutil.Max(latestBlock.Number-lp.finalityDepth, 0), nil
+	}
+
+	// If finality is enabled, we need to get the latest and finalized blocks.
+	blocks, err := lp.batchFetchBlocks(ctx, []string{rpc.LatestBlockNumber.String(), rpc.FinalizedBlockNumber.String()}, 2)
+	if err != nil {
+		return nil, 0, err
+	}
+	latest := blocks[0]
+	finalized := blocks[1]
+	return latest, finalized.Number, nil
+}
+
 // Find the first place where our chain and their chain have the same block,
 // that block number is the LCA. Return the block after that, where we want to resume polling.
-func (lp *logPoller) findBlockAfterLCA(ctx context.Context, current *evmtypes.Head) (*evmtypes.Head, error) {
+func (lp *logPoller) findBlockAfterLCA(ctx context.Context, current *evmtypes.Head, lastFinalizedBlockNumber int64) (*evmtypes.Head, error) {
 	// Current is where the mismatch starts.
 	// Check its parent to see if its the same as ours saved.
 	parent, err := lp.ec.HeadByHash(ctx, current.ParentHash)
 	if err != nil {
 		return nil, err
 	}
+	// If finalityTag is not used, we need to compute lastFinalizedBlockNumber based on the configured finalityDepth and the parent block
```

Comment:
@connorwstein I need your eyes to verify if that's true

Reply:
Yeah wait I believe this also applies to finality tags. For example - we crash during a reorg and come back up after a long downtime s.t. latestFinalizedBlockNumber is much further ahead. I believe we need to follow the same logic for latestFinalizedBlock as we do for latestBlock/currentBlock at the beginning of PollAndSaveLogs?